app-backend/tile/src/main/scala/TileIO.scala

```diff
@@ -0,0 +1,262 @@
+          Future {
+            Try {
+              S3CollectionLayerReader(store)
+                .query[SpatialKey, MultibandTile, TileLayerMetadata[SpatialKey]](LayerId(sceneId.toString, zoom))
+                .where(Intersects(extent))
+                .result
+                .stitch
+                .crop(extent)
+                .tile
+                .resample(256, 256)
+            } match {
+              case Success(tile) => Option(tile)
+              case Failure(e) =>
+                logger.error(s"Query layer $sceneId at zoom $zoom for $extent: ${e.getMessage}")
+                None
+            }
+          }
+        ))
+      )
+    )
+  }
+
+  /** Retrieve mosaic definitions -- usually the first step in constructing a mosaic
+    * for a project
+    *
+    * @param projectId project to request mosaic definition for
+    * @param polygonOption optional polygon to restrict mosaid definitions for
+    * @return
+    */
+  def getMosaicDefinition(projectId: UUID, polygonOption: Option[Projected[Polygon]]): OptionT[Future, Seq[MosaicDefinition]] = {
+    logger.debug(s"Reading mosaic definition (project: $projectId")
+    OptionT(timedFuture("get-mosaic-definition")(ScenesToProjects.getMosaicDefinition(projectId, polygonOption)))
+  }
+
+  /** Retrieve mosaic definitions -- usually the first step in constructing a mosaic
+    * for a project
+    *
+    * @param projectId project to request mosaic definition for
+    * @return
+    */
+  def getMosaicDefinition(projectId: UUID): OptionT[Future, Seq[MosaicDefinition]] = {
+    getMosaicDefinition(projectId, None)
+  }
+
+  /** Get tile of scene for a zoom level and spatial key
+    *
+    * @param sceneId ID of scene
+    * @param zoom level to request
+    * @param key spatial key of tile
+    * @return
+    */
+  def getSceneTile(sceneId: UUID, zoom: Int, key: SpatialKey): OptionT[Future, MultibandTile] = {
+    val cacheKey = s"tile-$sceneId-$zoom-${key.col}-${key.row}"
+    OptionT(rfCache.caching(cacheKey, doCache = cacheConfig.layerTile.enabled)(
+      timedFuture("s3-tile-request")(Future {
+        val reader = new S3ValueReader(store).reader[SpatialKey, MultibandTile](LayerId(sceneId.toString, zoom))
+        Try {
+          reader.read(key)
+        } match {
+          case Success(tile) => tile.some
+          case Failure(e: ValueNotFoundError) => None
+          case Failure(e) =>
+            logger.error(s"Reading layer $sceneId at $key: ${e.getMessage}")
+            None
+        }
+      })
+    ))
+  }
+
+  /** Get tile of scene given an ID, zoom, column, and row
+    *
+    * @param sceneId ID of scene to request tile for
+    * @param zoom level to request
+    * @param col x coordinate of tile
+    * @param row y coordinate of tile
+    * @return
+    */
+  def getSceneTileZXY(sceneId: UUID, zoom: Int, col: Int, row: Int): OptionT[Future, MultibandTile] = {
```

Comment:
Nitpick but the name should reveal that over-zooming will happen here vs `getSceneTile`.